src/GHCMod.hs

```diff
@@ -413,9 +436,19 @@ globalCommands (cmd:_) = case cmd of
     _ | cmd == "version" || cmd == "--version" -> Just progVersion
     _                                          -> Nothing
 
+getFileSourceFromStdin :: IO String
+getFileSourceFromStdin = do
+  let loop' acc = do
+        line <- getLine
+        if not (null line) && last line == '\EOT'
+        then return $ acc ++ init line
+        else loop' (acc++line++"\n")
+  loop' ""
+
 ghcCommands :: IOish m => [String] -> GhcModT m ()
 ghcCommands []         = fatalError "No command given (try --help)"
 ghcCommands (cmd:args) = do
+    loadMappedFiles
```

Comment:
Having this as part of run* functions wouldn't be wise. Essentially, this function only marshalls files specified on command line into a mutable structure. Rest of the code doesn't work with CLI options in any way (since well, neither ghc-modi, nor library has the luxury of CLI arguments)

That said, this could probably be more elegant. I'll see what I can do.